### PR TITLE
Adding a rigorous test for caching

### PR DESCRIPTION
**Pull request by chudur-budur**
This added unit-test preform different permutative caching operations and checks the validity of state of the cache.

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to an issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: numba_dpex/tests/kernel_tests/test_caching.py [hunk 1]
```diff
@@ -2,19 +2,92 @@
...
+
+    states = []
+    for i in range(4):
+        cache.put(i, alphabet[i])
+        tail_key = cache.get(cache.tail.key)
+        head_key = cache.get(cache.head.key)
+        states.append((cache, tail_key, head_key))
+    assert (
+        str(states)
+        == "[({(2: c), (1: b), (3: d), (0: a)}, 'a', 'a'), ({(2: c), (1: b), (3: d), (0: a)}, 'b', 'a'), ({(2: c), (1: b), (3: d), (0: a)}, 'c', 'b'), ({(2: c), (1: b), (3: d), (0: a)}, 'd', 'a')]"
```
Comment (diptorupd) [inline numba_dpex/tests/kernel_tests/test_caching.py:38]: Please reformat to fit 80 chars

**Review by adarshyoga — APPROVED** [review]
Much needed! One suggestion: Naming the test case "test_LRUCache" may be more appropriate since it specifically tests LRU ordering only. It will prevent someone from inadvertently using this test case to test a different caching mechanism.

File: numba_dpex/tests/kernel_tests/test_caching.py (hunk 1 above)
Comment (chudur-budur) [reply numba_dpex/tests/kernel_tests/test_caching.py]: Done.

**Comment by chudur-budur** [comment]
> Much needed! One suggestion: Naming the test case "test_LRUCache" may be more appropriate since it specifically tests LRU ordering only. I will prevent someone from inadvertently using this test case to test a different caching mechanism.

Done!

**Pull request merged**